src/orquestra/sdk/_ray/_build_workflow.py

```diff
@@ -198,69 +197,47 @@ def _ray_remote(*inner_args, **inner_kwargs):
             with _markers.printed_task_markers(
                 wf_run_id=wf_run_id, task_inv_id=task_inv_id
             ):
-                try:
-                    if project_dir is not None:
-                        dispatch.ensure_sys_paths([str(project_dir)])
-
-                    if user_fn_ref is None:
-                        serialization = False
-                        user_fn = _aggregate_outputs
-                    else:
-                        serialization = True
-                        user_fn = _locate_user_fn(user_fn_ref)
-
-                    wrapped = ArgumentUnwrapper(
-                        user_fn=user_fn,
-                        args_artifact_nodes=args_artifact_nodes,
-                        kwargs_artifact_nodes=kwargs_artifact_nodes,
-                        deserialize=serialization,
-                    )
+                if project_dir is not None:
+                    dispatch.ensure_sys_paths([str(project_dir)])
+
+                if user_fn_ref is None:
+                    serialization = False
+                    user_fn = _aggregate_outputs
+                else:
+                    serialization = True
+                    user_fn = _locate_user_fn(user_fn_ref)
+
+                wrapped = ArgumentUnwrapper(
+                    user_fn=user_fn,
+                    args_artifact_nodes=args_artifact_nodes,
+                    kwargs_artifact_nodes=kwargs_artifact_nodes,
+                    deserialize=serialization,
+                )
 
-                    wrapped_return = wrapped(*inner_args, **inner_kwargs)
+                wrapped_return = wrapped(*inner_args, **inner_kwargs)
 
-                    packed: responses.WorkflowResult = (
+                packed: responses.WorkflowResult = (
+                    serde.result_from_artifact(wrapped_return, ir.ArtifactFormat.AUTO)
+                    if serialization
+                    else wrapped_return
+                )
+                unpacked: t.Tuple[responses.WorkflowResult, ...]
+
+                if n_outputs is not None and n_outputs > 1:
+                    unpacked = tuple(
                         serde.result_from_artifact(
-                            wrapped_return, ir.ArtifactFormat.AUTO
+                            wrapped_return[i], ir.ArtifactFormat.AUTO
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏This could be breaking. Have you checked how this works with CE and stored Ray data?

Reply:
I'm glad you raised this but nope. All I've done was dedent the block. Git diff heuristic messed things up, please a take look at the side-by-side view.

I was also concerned by this when I was doing a self-review. Actually, this was the original reason why I've spent this morning refactoring this code (e.g. extracting the crucial bit into a separate function), and it's how I've discovered the bug I mentioned during standup (and you promised to take a look at later today!).